dashboard/src/actions/repos.ts

```diff
@@ -208,12 +208,25 @@ export const fetchRepos = (
       if (!otherNamespaces || !otherNamespaces.length) {
         dispatch(receiveRepos(repos.items));
       } else {
-        let totalRepos = repos.items;
+        const totalRepos = repos.items;
         await Promise.all(
           otherNamespaces.map(async otherNamespace => {
             dispatch(requestRepos(otherNamespace));
             const otherRepos = await AppRepository.list(currentCluster, otherNamespace);
-            totalRepos = totalRepos.concat(otherRepos.items);
+            // Avoid addiing duplicated repos: if two repos have the same uid, skip one.
+            otherRepos.items.forEach((otherRepo: IAppRepository) => {
+              if (otherRepo?.metadata?.uid) {
+                const index = totalRepos.findIndex((r: IAppRepository) =>
+                  r?.metadata?.uid ? otherRepo.metadata.uid === r?.metadata?.uid : -1,
+                );
+                if (index === -1) {
+                  totalRepos.push(otherRepo);
+                }
+              } else {
+                // if no uid, add it anyway
+                totalRepos.push(otherRepo);
+              }
+            });
```

Comment:
I still think this should not be needed but you can replace all this code simply doing at the end:

```
totalRepos = uniqBy(totalRepos, "metadata.uid")
```

(uniqBy from lodash)

Reply:
I'll do it in another PR, but... lodash is such a good thing for helping with these basic functions. I should think about it more often!